addons/survey/controllers/main.py

```diff
@@ -150,6 +150,8 @@ def _redirect_with_error(self, access_data, error_key):
             return request.render("survey.survey_auth_required", {'survey': survey_sudo, 'redirect_url': redirect_url})
         elif error_key == 'answer_deadline' and answer_sudo.access_token:
             return request.render("survey.survey_closed_expired", {'survey': survey_sudo})
+        elif error_key in ['answer_wrong_user', 'token_wrong']:
```

Comment:
I think token_wrong shouldn't be here. If the token is wrong and we're not in 'invite only' mode the code should already be generating a new token.

Reply:
Hi @thiryr 

But it will be the only case when new token is not being generated.

Thanks

Reply:
I don't think it can ever happen in start_survey because it always bypasses the ``answer_token`` check, as it doesn't matter as long as we can generate a new one. Whereas ``answer_wrong_user`` can happen even without the ``answer_token`` check.

I think the only time this would actually happen is when someone tries to join a survey with a non-start link like ``http://[addr]/survey/[survey-token]/[answer-token]`` but the answer doesn't exist in DB or never existed. In which case I don't think the error message is relevant.

Either way I don't think the error message describes what ``token_wrong`` means properly. Maybe it could be a separate message like "We could not find your previous answers". Or redirect to the start link without ``answer_token``. Or both :)

Reply:
Hi,

I checked again for survey login with another user than the shared one.

Initially, it goes to the `/survey/start/<string:survey_token>` route and since it has an answer token linked to it (as query param), it doesn't change the `answer_from_cookie` value and it will remain False.

Now, we got `answer_wrong_user` validity code and `answer_from_cookie` is also false, it won't create another access for the logged-in user.

and hence, it will continue with the `answer_wrong_user` and redirect the user to the `/` page.
And also, This is the starting endpoint when user got the invite.

This is the Initial flow, if we want to create a new session for the logged-in user then we have to remove the condition of `answer_from_cookie` 

Please correct me if I am missing anything in this.

Thanks

Reply:
Ah yes actually I was misunderstood something in that flow, thanks for the explanation and sorry about that ^^'

I'm still not convinced they should receive the same error message though. I think if they miss a character when copying the link, or if there's an issue with the database, they shouldn't be told they were not invited. Could we just add a similar error template that says "The link you have provided does not match any survey entry" or something?

Maybe to see with lna as it's more a functional problem.